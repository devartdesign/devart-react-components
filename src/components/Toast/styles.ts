import { StyleSheet } from 'aphrodite/no-important';
import { hexToRgba } from 'devart-utils/dist/utils';
import { Color } from 'devart-utils/dist/models/general';

export default StyleSheet.create({
  container: {
    overflow: 'hidden',
    border: '1px solid rgba(0,0,0,.1)',
    boxShadow: '0 4px 12px rgba(0,0,0,.1)',
    borderRadius: 4,
    maxWidth: 300,
    minWidth: 300,
    backgroundClip: 'padding-box'
  },
  title: {
    textTransform: 'capitalize',
    fontWeight: 'bold'
  },
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    padding: '4px 12px',
    borderBottom: '1px solid rgba(0,0,0,.05)',
    lineHeight: 1.8,
    textTransform: 'capitalize',
    color: Color.DARK_GRAY,
    backgroundColor: hexToRgba(Color.WHITE, 85),
    backgroundClip: 'padding-box'
  },
  button: {
    appearance: 'none',
    padding: 0,
    paddingLeft: 10,
    border: 0,
    backgroundColor: 'transparent'
  },
  body: {
    padding: 12,
    textTransform: 'capitalize',
    color: Color.WHITE
  },
  success: {
    backgroundColor: Color.GREEN
  },
  warning: {
    backgroundColor: Color.FAVORITE
  },
  info: {
    backgroundColor: Color.SECONDARY
  },
  danger: {
    backgroundColor: Color.NONARY
  }
});
