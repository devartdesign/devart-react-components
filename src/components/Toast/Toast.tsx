import React, { memo } from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export enum ToastType {
  SUCCESS = 'success',
  WARN = 'warning',
  ERROR = 'danger',
  INFO = 'info'
}

export interface IToastProps {
  title: string;
  description: string;
  type: ToastType;
  onClose: () => void;
}

const Toast = ({ title, description, type, onClose }: IToastProps) => {
  return (
    <div key="Toast" className={css(styles.container, styles[type])}>
      <div className={css(styles.header)}>
        <strong>{title}</strong>
        <button data-testid="close-toast-btn" className={css(styles.button)} onClick={onClose}>
          <small>x</small>
        </button>
      </div>
      <div className={css(styles.body)}>{description}</div>
    </div>
  );
};

export default memo(Toast);
