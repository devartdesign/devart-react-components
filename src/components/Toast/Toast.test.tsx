import * as React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Toast, { IToastProps, ToastType } from './Toast';

describe('Toast', () => {
  let props: IToastProps;
  let Component;

  beforeEach(() => {
    props = {
      title: 'title',
      description: 'description',
      type: ToastType.SUCCESS,
      onClose: jest.fn()
    };
    Component = shallow<Toast>(<Toast {...props} />);
  });

  it('should render', () => {
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should onClose', () => {
    Component.find('button').simulate('click');
    expect(props.onClose).toBeCalled();
  });
});
