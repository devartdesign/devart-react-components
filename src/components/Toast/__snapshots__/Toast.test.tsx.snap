// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render 1`] = `
.container-success {
  overflow: hidden;
  border: 1px solid rgba(0,0,0,.1);
  box-shadow: 0 4px 12px rgba(0,0,0,.1);
  border-radius: 4px;
  max-width: 300px;
  min-width: 300px;
  background-clip: padding-box;
  background-color: #3AC6AB;
}

.header {
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 4px 12px;
  border-bottom: 1px solid rgba(0,0,0,.05);
  line-height: 1.8;
  text-transform: capitalize;
  color: #646565;
  background-color: #e9e9e9;
  background-clip: padding-box;
}

.button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0px;
  padding-left: 10px;
  border: 0px;
  background-color: transparent;
}

.body {
  padding: 12px;
  text-transform: capitalize;
  color: #FFF;
}

<div
  className="container-success"
>
  <div
    className="header"
  >
    <strong>
      title
    </strong>
    <button
      className="button"
      data-testid="close-toast-btn"
      onClick={[MockFunction]}
    >
      <small>
        x
      </small>
    </button>
  </div>
  <div
    className="body"
  >
    description
  </div>
</div>
`;
